src/lib/mdns/minimal/Server.cpp

```diff
@@ -159,6 +168,54 @@ CHIP_ERROR ServerBase::DirectSend(chip::System::PacketBuffer * data, const chip:
     return CHIP_ERROR_NOT_CONNECTED;
 }
 
+CHIP_ERROR ServerBase::BroadcastSend(chip::System::PacketBuffer * data, uint16_t port, chip::Inet::InterfaceId interface)
+{
+    for (size_t i = 0; i < mEndpointCount; i++)
+    {
+        EndpointInfo * info = &mEndpoints[i];
+
+        if (info->udp == nullptr)
+        {
+            continue;
+        }
+
+        if ((info->udp->GetBoundInterface() != interface) && (info->udp->GetBoundInterface() != INET_NULL_INTERFACEID))
+        {
+            continue;
+        }
+
+        // data may be sent over multiple packets. Keep the one ref active all the time
```

Comment:
This seems to send to one interface, but the code below seems to imply that we must upref to send on more?

Reply:
The same interface is used for both client and server (server may be a bad name really ... but I have nothing better ... instance?).

The logic here is for 'broadcast a query everywhere' like 'find nodeid on any interface' in which case the packet is broadcasted over several interfaces/sends.

The code is awkward. It will become slightly less awkward in the 'send query' PR.